### PR TITLE
show debug hit in design view

### PR DESCRIPTION
**Pull request by pahans**

### CODE REVIEW

File: modules/web/js/ballerina/ast/node.js [hunk 1]
```diff
@@ -436,6 +436,14 @@ class ASTNode extends EventChannel {
         this.isBreakpoint = false;
     }
 
+    addDebugHit() {
+        this.setAttribute('isDebugHit', true);
+    }
+
+    removeDebugHit() {
```
Comment (hemikak) [inline modules/web/js/ballerina/ast/node.js]: Better to have doc comments here.

File: modules/web/js/ballerina/components/statement-decorator.jsx [hunk 2]
```diff
@@ -155,6 +155,10 @@ class StatementDecorator extends React.Component {
 		actionBbox.h = DesignerDefaults.actionBox.height;
 		actionBbox.x = bBox.x + ( bBox.w - actionBbox.w) / 2;
 		actionBbox.y = bBox.y + bBox.h + DesignerDefaults.actionBox.padding.top;
+		let statementRectClass = "statement-rect";
+		if(model.isDebugHit) {
```
Comment (hemikak) [inline modules/web/js/ballerina/components/statement-decorator.jsx]: Formatting issue between the if keyword and openning bracket.

File: modules/web/js/ballerina/visitors/find-debug-hit-visitor.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import ASTVisitor from './ast-visitor';
+
+class FindDebugHitVisitor extends ASTVisitor {
```
Comment (hemikak) [inline modules/web/js/ballerina/visitors/find-debug-hit-visitor.js]: Need to have class comments here.

Comment (pahans) [inline modules/web/js/ballerina/visitors/find-debug-hit-visitor.js]: fixed with https://github.com/ballerinalang/composer/pull/2011

File: modules/web/js/ballerina/components/statement-decorator.jsx (hunk 2 above)
Comment (pahans) [inline modules/web/js/ballerina/components/statement-decorator.jsx]: fixed with https://github.com/ballerinalang/composer/pull/2012

File: modules/web/js/ballerina/ast/node.js (hunk 1 above)
Comment (pahans) [inline modules/web/js/ballerina/ast/node.js]: fixed with https://github.com/ballerinalang/composer/pull/2013

**Pull request merged**